HW2/P5/physics.pyx

```diff
@@ -113,18 +140,36 @@ cpdef update(FLOAT[:, ::1] XY,
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads, with static
         # scheduling.
-        for i in range(count):
-            sub_update(XY, V, R, i, count, Grid, grid_spacing)
+        for i in prange(count, num_threads=num_threads, schedule='static', chunksize=chunksize):
+            sub_update(XY, V, R, i, count, Grid, gridsize, grid_spacing, locks)
 
         # update positions
         #
         # SUBPROBLEM 1: parallelize this loop over 4 threads (with static
         #    scheduling).
         # SUBPROBLEM 2: update the grid values.
-        for i in range(count):
+
+        # set grid value to -1
+        for i in prange(gridsize):
+            for j in range(gridsize):
+                Grid[i, j] = -1
+
```

Comment:
Why do you need to clear all Grid? Just removing from the old place should be enough.